### PR TITLE
Refactor client creation/usage

### PR DESCRIPTION
**Pull request by coreydaley**
Refactor client creation/usage between the controller and the generator
to reuse already created clients instead of creating so many new ones

### CODE REVIEW

**Comment by coreydaley** [comment]
/retest
unrelated failures/flakes

**Comment by coreydaley** [comment]
@adambkaplan bump

File: pkg/operator/controller.go [hunk 1]
```diff
@@ -12,15 +12,20 @@ import (
...
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/util/workqueue"
 	"k8s.io/klog"
 
 	operatorapi "github.com/openshift/api/operator/v1"
 	configset "github.com/openshift/client-go/config/clientset/versioned"
+	configsetv1 "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
 	configinformers "github.com/openshift/client-go/config/informers/externalversions"
 	routeset "github.com/openshift/client-go/route/clientset/versioned"
+	routesetv1 "github.com/openshift/client-go/route/clientset/versioned/typed/route/v1"
```
Comment (adambkaplan) [inline pkg/operator/controller.go]: Why do we have unversioned and versioned ClientSets?

Comment (coreydaley) [reply pkg/operator/controller.go]: The informers needed the versioned ones, and the generators needed the unversioned ones

**Comment by adambkaplan** [comment]
/lgtm

As a follow-up the generators should use versioned clientsets (if possible)

**Pull request merged**